server/interpreter.js

```diff
@@ -5430,10 +5433,21 @@ stepFuncs_['Call'] = function (thread, stack, state, node) {
         func.construct(this, thread, state, args) :
         func.call(this, thread, state, state.info_.this, args);
     if (r instanceof Interpreter.FunctionResult) {
-      if (r === Interpreter.FunctionResult.CallAgain) {
-        state.step_ = 0;
-      }
-      return;  // N.B. SEE NOTE ABOVE!
+      switch (r) {
+        case Interpreter.FunctionResult.AwaitValue:
+          return;
+        case Interpreter.FunctionResult.Block:
+          thread.status = Interpreter.Thread.Status.BLOCKED;
+          return;
+        case Interpreter.FunctionResult.CallAgain:
+          state.step_ = 0;
+          return;
+        case Interpreter.FunctionResult.Sleep:
+          thread.status = Interpreter.Thread.Status.SLEEPING;
+          return;
+        default:
+          throw Error('Unknown FunctionResult??');
```

Comment:
This is a great opportunity for an interbang.

Reply:
Notation somewhat inspired by [Chess annotation symbols](https://en.wikipedia.org/wiki/Chess_annotation_symbols).  "Blunder" is definitely more appropriate than "dubious move" in this case.